### ISSUE TITLE
exclude stable currencies pairs

### ISSUE THREAD

**Issue by orehunt** [freqtrade/freqtrade#2783]
There should be a _best effort_ option to exclude stablecoins to stablecoins pairs like `USDC/USDT`.
It could be done with 
- a mantained list of tickers, this avoids computation by just appending matches to the blacklist
- a preset signal that is very good at spotting flatlines...you would still compute all of them though

**Comment by xmatthias** [comment]
I'm not sure about this ... it's asking for a hardcoded list of currencies to be added to the blacklist...

Personally, I think it depends on a strategy, and freqtrade should not limit users in trying to trade these (even though i may doubt the possibility to be profitable ...).

* Why is it included in your pairlist anyway? 
* and if you're using VolumePairlist - wouldn't adding them to the blacklist suffice to mitigate this?

The best thing i can think of is to verify on the exchange if it's a valid stake currency and exclude based on that - but looking at binance, that would also eliminate ETH/USDT, BTC/USDT and XRP/USDT pairs, where it's most likely not desired.

In short - to me this is not something we should do in freqtrade - but it's part of a user configuration.

**Comment by hroff-1902** [comment]
I'm also not sure about this.

For example, there are exchanges with EUR as stake currency. On those exchanges, USDT/EUR, USDC/EUR are okay to trade... The stable pairs can easily be included into backlist for each particular stake currency and each particular exchange, imho...

As a possible enhancement -- there can be an additional PairList based on volatility implemented, which will select out such the pairs...

**Comment by orehunt** [comment]
> Personally, I think it depends on a strategy, and freqtrade should not limit users in trying to trade these (even though i may doubt the possibility to be profitable ...).
Well if ever there was such a list it would be always an option to toggle on/off like `exclude_stablecoins`
> wouldn't adding them to the blacklist suffice to mitigate this?
Yes but that it is something you would have to do for each exchange and it would have to be something you always have to look out for
> that would also eliminate ETH/USDT, BTC/USDT and XRP/USDT pairs, where it's most likely not desired.
The matching algo would simply match for start and end of a pair:
- if PAIR starts with any of (stablecoins) and ends with any of (stablecoins) -> blacklist
> For example, there are exchanges with EUR as stake currency. On those exchanges, USDT/EUR, USDC/EUR are okay to trade... The stable pairs can easily be included into backlist for each particular stake currency and each particular exchange, imho...
That's the trickiest part I think, because a pair between stablecoins pegged to different fiats would be excluded, so you would have to group stablecoins to their relative pegs :(

**Comment by hroff-1902** [comment]
In principle, we do not require the pairs in the blacklist to be of the used stake currency (as required and checked for the currencies in the whitelist), so it could be a sample blacklist (probably set in a separate sample config file, illustrating this, which can be used for copy-pasting pairs into the working config):
```
        "pair_blacklist": [
            "USDT/USD",
            "USDC/USD",
            "TUSD/USD",
            "USD/USDT",
            "USDC/USDT",
            "TUSD/USDT",
            "USDT/USDC",
            "USD/USDC",
            "TUSD/USDC",
...
        ],
```

**Comment by hroff-1902** [comment]
@orehunt do you propose the following in the config?

```
"stablecoins_list": [
    "USD", "USDT", "USDC", "TUSD", // Add more if any
],
"exclude_stablecoins": true,
```

**Comment by orehunt** [comment]
@hroff-1902 Since currently you have to specify all the pairs in the whitelist specifying a list for stablecoins would be redundant. If the whitelist had globbing e.g. `{ "*/USDT" }` then a stablecoins list would make sense.

**Comment by xmatthias** [comment]
> If the whitelist had globbing e.g. { "*/USDT" } then a stablecoins list would make sense.

Well technically, VolumePairlist does exactly this - you specify a stake-currency - and volumepairlist then selects all coins (if you set the pair-limit high enough, you'll get all coins with that stake-currency).

**Comment by orehunt** [comment]
@xmatthias but I don't think it works with backtesting

**Comment by hroff-1902** [comment]
> @xmatthias but I don't think it works with backtesting

Simply add all the pairs you want to ignore into the static blacklist. Are there many pairs?
